art/estimators/object_detection/pytorch_detection_transformer.py

```diff
@@ -0,0 +1,998 @@
+            or self._input_shape[1] > self.MAX_IMAGE_SIZE
+            or self._input_shape[2] < self.MIN_IMAGE_SIZE
+            or self.input_shape[2] > self.MAX_IMAGE_SIZE
+        ):
+            resized_imgs = []
+            if isinstance(x, torch.Tensor):
+                x = T.Resize(size=(height, width))(x)
+            else:
+                for i in x:
+                    resized = cv2.resize(
+                        i.transpose(1, 2, 0),
+                        dsize=(height, width),
+                        interpolation=cv2.INTER_CUBIC,
+                    )
+                    resized = resized.transpose(2, 0, 1)
+                    resized_imgs.append(resized)
+                x = np.array(resized_imgs)
+
+        elif self._input_shape[1] != self._input_shape[2]:
+            rescale_dim = max(self._input_shape[1], self._input_shape[2])
+            resized_imgs = []
+            if isinstance(x, torch.Tensor):
+                x = T.Resize(size=(rescale_dim, rescale_dim))(x)
+            else:
+                for i in x:
+                    resized = cv2.resize(
+                        i.transpose(1, 2, 0),
+                        dsize=(rescale_dim, rescale_dim),
+                        interpolation=cv2.INTER_CUBIC,
+                    )
+                    resized = resized.transpose(2, 0, 1)
+                    resized_imgs.append(resized)
+                x = np.array(resized_imgs)
+
+        targets: List[Any] = []
+        if y is not None:
+            if isinstance(y[0]["boxes"], torch.Tensor):
+                for target in y:
+                    assert isinstance(target["boxes"], torch.Tensor)
+                    assert isinstance(target["labels"], torch.Tensor)
+                    assert isinstance(target["scores"], torch.Tensor)
+                    cxcy_norm = revert_rescale_bboxes(target["boxes"], (self.input_shape[2], self.input_shape[1]))
+                    targets.append(
+                        {
+                            "labels": target["labels"].type(torch.int64).to(self.device),
+                            "boxes": cxcy_norm.to(self.device),
+                            "scores": target["scores"].type(torch.float).to(self.device),
+                        }
+                    )
+            else:
+                for target in y:
+                    tensor_box = torch.from_numpy(target["boxes"])
+                    cxcy_norm = revert_rescale_bboxes(tensor_box, (self.input_shape[2], self.input_shape[1]))
+                    targets.append(
+                        {
+                            "labels": torch.from_numpy(target["labels"]).type(torch.int64).to(self.device),
+                            "boxes": cxcy_norm.to(self.device),
+                            "scores": torch.from_numpy(target["scores"]).type(torch.float).to(self.device),
+                        }
+                    )
+
+        return x, targets
+
+
+class NestedTensor:
```

Comment:
We need to mention the license of these functions in their docstrings.

Reply:
Added reference to Apache 2.0